src/test/java/org/openrewrite/staticanalysis/TernaryOperatorsShouldNotBeNestedTest.java

```diff
@@ -0,0 +1,1315 @@
+                  """,
+                """
+                  class Test {
+                    public String determineSomething(int a) {
+                        return switch (a) {
+                            case 1 -> "one";
+                            case 2 -> "two";
+                            default -> "other";
+                        };
+                    }
+                  }
+                  """
+              )
+            );
+        }
+
+        @Test
+        void doReplaceNestedOrTernaryWithSwitchExpressionInvertedEquals() {
+            rewriteRun(
+              //language=java
+              java(
+                """
+                  class Test {
+                    public String determineSomething(String a, String b) {
+                      return a.equals("a") ? "a" : a.equals("b") ? b : "nope";
+                    }
+                  }
+                  """,
+                """
+                  class Test {
+                    public String determineSomething(String a, String b) {
+                      return switch (a) {
+                          case "a" -> "a";
+                          case "b" -> b;
+                          default -> "nope";
+                      };
+                    }
+                  }
+                  """
+              )
+            );
+        }
+
+        @Test
+        void doReplaceNestedOrTernaryWithSwitchExpressionNullSafeEquals() {
+            rewriteRun(
+              //language=java
+              java(
+                """
+                  import java.util.Objects;
+                  class Test {
+                    public String determineSomething(String a, String b) {
+                      return Objects.equals(a, "a") ? "a" : Objects.equals(a, "b") ? b : "nope";
+                    }
+                  }
+                  """,
+                """
+                  class Test {
+                    public String determineSomething(String a, String b) {
+                      return switch (a) {
+                          case "a" -> "a";
+                          case "b" -> b;
+                          default -> "nope";
+                      };
+                    }
+                  }
+                  """
+              )
+            );
+        }
+
+        @Test
+        void doReplaceNestedOrTernaryWithSwitchExpressionNullSafeEqualsInverted() {
+            rewriteRun(
+              //language=java
+              java(
+                """
+                  import java.util.Objects;
+                  class Test {
+                    public String determineSomething(String a, String b) {
+                      return Objects.equals("a", a) ? "a" : Objects.equals("b", a) ? b : "nope";
+                    }
+                  }
+                  """,
+                """
+                  class Test {
+                    public String determineSomething(String a, String b) {
+                      return switch (a) {
```

Comment:
Don't we lose null safety here?

Reply:
You are right, I did not know switch needs a nonNull arg.
An easy way around this is to skip `Objects.equals`, alternatively we can add a null check and early exit with the "default" case. What do you think?

Reply:
I'd say we can skip `Objects.equals` now and revisit when Java 21 is out, as Java 18+ allows for null in switch.